$OpenBSD: patch-taglib_mp4_mp4tag_cpp,v 1.1 2012/10/16 13:06:13 dcoppa Exp $

Fix a bug in updating MP4 tags
(upstream git commit a9df3e48f7b59bc27e115e5b0ad057a642b6fac3)

--- taglib/mp4/mp4tag.cpp.orig	Tue Oct 16 13:49:35 2012
+++ taglib/mp4/mp4tag.cpp	Tue Oct 16 13:50:12 2012
@@ -570,7 +570,7 @@ MP4::Tag::updateOffsets(long delta, long offset)
         atom->offset += delta;
       }
       d->file->seek(atom->offset + 9);
-      ByteVector data = d->file->readBlock(atom->offset - 9);
+      ByteVector data = d->file->readBlock(atom->length - 9);
       unsigned int flags = (ByteVector(1, '\0') + data.mid(0, 3)).toUInt();
       if(flags & 1) {
         long long o = data.mid(7, 8).toLongLong();
